.github/workflows/govulncheck.yml

```diff
@@ -0,0 +1,100 @@
+# This workflow is meant to run govulncheck on all the branches
+# that are containing a maintained version of OpenTofu.
+# For more considerations about this, check this PR: https://github.com/opentofu/opentofu/pull/2600
+#
+# This will attempt to create an issue with the results for each scanned version
+# in case any vulnerability is found. If the issue for the scanned branch already exists
+# the run will just print the scan output and will leave the issue untouched.
+#
+# This is meant to run _only_ from the main branch, on a scheduled manner.
+# All the other branches will be scanned directly by the run triggered from the main branch.
+
+name: Govulncheck
+
+on:
+  schedule:
+    - cron: '42 3 * * SUN'
+  workflow_dispatch: {}
+
+jobs:
+  gather-branches:
+    name: Collect the branches that needs to be scanned
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    env:
+      GH_TOKEN: ${{ github.token }}
+    outputs:
+      branches: ${{ steps.branches.outputs.branches }}
+    steps:
+      - id: branches
+        run: |
+          ## This is using GitHub API and we manually filter out the versions that are not supported anymore
+          # gh api "repos/opentofu/opentofu/branches?protected=true" --paginate | tee out 
+          # cat out | jq '.[] | select(.name != "v1.6") | [.name]' | jq -sc 'add' | tee branches
+          # echo "branches=$(cat branches)">> "$GITHUB_OUTPUT"
```

Comment:
This is a first approach where I used the GH API to get all the protected branches (which are only the ones that are for the versions released of OpenTofu) and out of those exclude the the unmaintained versions.
Pros:
* is using a reliable and well-known API that we can fully trust.

Cons:
* manual maintenance. Quite ugly to update this.

Reply:
The update on the site is done automatically as seen [here](https://github.com/endoflife-date/release-data/tree/main?tab=readme-ov-file).
![image](https://github.com/user-attachments/assets/01e8497d-1e49-4104-9ed8-21c39e9a4928)

Reply:
We are keeping only the hardcoded strategy to hold the control on what branches we run the flow against.
See [this](https://github.com/opentofu/opentofu/pull/2636#discussion_r2033011305) suggestion.